### ISSUE TITLE
Undefined variable: b:syntastic_changedtick

### ISSUE THREAD

**Issue by yriveiro** [scrooloose/syntastic#1865]
Every time I save a file I have this message error:

```
Error detected while processing function <SNR>50_BufWritePostHook[4]..<SNR>50_UpdateErrors:
line   46:
E121: Undefined variable: b:syntastic_changedtick
E15: Invalid expression: [a:buf, b:syntastic_changedtick]
```

- My vim version: stable 7.4.2152
- Syntactic: commit a0f5b72f14fba99f91b07580d0a11c96900fb242

**Comment by lcd047** [comment]
Please upgrade to the latest master HEAD and try again.

**Comment by lcd047** [comment]
Please see #1862.  Not the same problem, but still not syntastic's problem, either.

**Comment by yriveiro** [comment]
If the problem is vim related the patch is only for 7.4.2299 and I'm in 7.4.2152, should I upgrade vim?

**Comment by lcd047** [comment]
The problem is likely to be triggered by a third-party plugin, and it's also likely to be related to an `autocmd`.  Find out what plugin triggers it, and I _might_ be able to tell you what's going on.

**Comment by yriveiro** [comment]
I'm not a vim expert, how can I debug this?

This is the list of third-party plugins I have installed:

Plug 'tpope/vim-sensible'
Plug 'scrooloose/nerdtree'
Plug 'tpope/vim-fugitive'
Plug 'airblade/vim-gitgutter'
Plug 'scrooloose/syntastic'
Plug 'tpope/vim-surround'
Plug 'ctrlpvim/ctrlp.vim'
Plug 'bling/vim-airline'
Plug 'valloric/youcompleteme'
Plug 'sirver/ultisnips'
Plug 'honza/vim-snippets'
Plug 'roosta/srcery'
Plug 'xuyuanp/nerdtree-git-plugin'
Plug 'mileszs/ack.vim'
Plug 'terryma/vim-multiple-cursors'
Plug 'editorconfig/editorconfig-vim'
Plug 'xolox/vim-session'
Plug 'xolox/vim-misc'
Plug 'scrooloose/nerdcommenter'

" Loaded when specific file type is opened
Plug 'stephpy/vim-php-cs-fixer', { 'for': 'php' }
Plug 'rust-lang/rust.vim', {'for': 'rust'}

**Comment by lcd047** [comment]
Disable the other plugins (i.e. comment out the `Plug` lines above) and check that syntastic works as expected.  Then enable them in turn, and see which one breaks syntastic.  Start with `xuyuanp/nerdtree-git-plugin`, IIRC it had problems with syntastic in the past and the problems were never fixed to this day.

It's a relatively short list, but if you want to optimise your work you can do binary search: comment out half of them, to find out if the offending one is in the first half or the second half etc.

**Comment by yriveiro** [comment]
@lcd047, you hit the spot, the problematic plugin is ```xuyuanp/nerdtree-git-plugin```

I did a full plugin upgrade this morning, before this all work with any issue ...

**Comment by lcd047** [comment]
This should be fixed in 557b52f.  Please upgrade to the current master HEAD and try again.

**Comment by yriveiro** [comment]
The problem was fixed.
